http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala

```diff
@@ -97,26 +98,25 @@ class ProxyService(
     .expireAfterWrite(proxyConfig.internalIdCacheExpiryTime.toSeconds, TimeUnit.SECONDS)
     .maximumSize(proxyConfig.internalIdCacheMaxSize)
     .build(
-      new CacheLoader[(GoogleProject, ClusterName), Future[Option[ClusterInternalId]]] {
-        def load(key: (GoogleProject, ClusterName)) = {
+      new CacheLoader[(GoogleProject, ClusterName), Option[ClusterInternalId]] {
+        def load(key: (GoogleProject, ClusterName)): Option[ClusterInternalId] = {
           val (googleProject, clusterName) = key
-          dbRef.inTransaction { dataAccess =>
-            dataAccess.clusterQuery.getActiveClusterInternalIdByName(googleProject, clusterName)
-          }
+          dbRef.inTransaction {
+            dbRef.dataAccess.clusterQuery.getActiveClusterInternalIdByName(googleProject, clusterName)
+          }.unsafeRunSync()
         }
       }
     )
 
   def getCachedClusterInternalId(googleProject: GoogleProject, clusterName: ClusterName)(
     implicit ev: ApplicativeAsk[IO, TraceId]
-  ): Future[ClusterInternalId] =
-    clusterInternalIdCache.get((googleProject, clusterName)).flatMap {
-      case Some(clusterInternalId) => Future.successful(clusterInternalId)
+  ): IO[ClusterInternalId] =
+    IO(clusterInternalIdCache.get((googleProject, clusterName))).flatMap {
```

Comment:
This might need `blocker.blockOn(IO(clusterInternalIdCache.get(…)))`. I remember being bitten in the past by this.